File: examples/health_advisor/lib/theme.dart
import 'package:flutter/material.dart';

/// Centralized theme for the Health Advisor app.
///
/// Green accent on true-black background, with confidence-level colors
/// for the badge system (green/orange/red).
class AppTheme {
  AppTheme._();

  // -- Core Colors --------------------------------------------------------

  /// True black background
  static const Color background = Color(0xFF000000);

  /// Near-black card surface
  static const Color surface = Color(0xFF121212);

  /// Slightly lighter surface for elevated elements
  static const Color surfaceVariant = Color(0xFF1E1E1E);

  /// Green primary accent
  static final Color accent = Colors.green.shade400;

  /// Darker green for secondary elements
  static final Color accentDark = Colors.green.shade700;

  // -- Text Colors --------------------------------------------------------

  /// Near-white primary text
  static const Color textPrimary = Color(0xFFF5F5F5);

  /// Muted secondary text
  static const Color textSecondary = Color(0xFF8A8A9A);

  /// Very muted hint text
  static const Color textTertiary = Color(0xFF5A5A6A);

  // -- UI Element Colors --------------------------------------------------

  /// Subtle border
  static const Color border = Color(0xFF2A2A2A);

  /// User chat bubble (green)
  static final Color userBubble = Colors.green.shade700;

  /// Assistant chat bubble
  static const Color assistantBubble = Color(0xFF1E1E1E);

  // -- Confidence Colors --------------------------------------------------

  /// High confidence (>0.7)
  static final Color confidenceHigh = Colors.green.shade400;

  /// Medium confidence (0.4-0.7)
  static final Color confidenceMedium = Colors.orange.shade400;

  /// Low confidence (<0.4)
  static final Color confidenceLow = Colors.red.shade400;

  /// Get confidence color based on score
  static Color confidenceColor(double score) {
    if (score > 0.7) return confidenceHigh;
    if (score > 0.4) return confidenceMedium;
    return confidenceLow;
  }

  // -- ThemeData ----------------------------------------------------------

  static ThemeData get themeData {
    return ThemeData(
      brightness: Brightness.dark,
      useMaterial3: true,
      scaffoldBackgroundColor: background,
      colorScheme: ColorScheme.dark(
        surface: surface,
        primary: accent,
        secondary: accentDark,
        onSurface: textPrimary,
        onPrimary: background,
        error: Colors.red.shade400,
      ),
      appBarTheme: const AppBarTheme(
        backgroundColor: background,
        foregroundColor: textPrimary,
        elevation: 0,
        centerTitle: true,
      ),
      elevatedButtonTheme: ElevatedButtonThemeData(
        style: ElevatedButton.styleFrom(
          backgroundColor: accent,
          foregroundColor: background,
          shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(12),
          ),
        ),
      ),
      inputDecorationTheme: InputDecorationTheme(
        filled: true,
        fillColor: surfaceVariant,
        border: OutlineInputBorder(
          borderRadius: BorderRadius.circular(12),
          borderSide: const BorderSide(color: border),
        ),
        enabledBorder: OutlineInputBorder(
          borderRadius: BorderRadius.circular(12),
          borderSide: const BorderSide(color: border),
        ),
        focusedBorder: OutlineInputBorder(
          borderRadius: BorderRadius.circular(12),
          borderSide: BorderSide(color: accent),
        ),
      ),
      snackBarTheme: const SnackBarThemeData(
        backgroundColor: surface,
        contentTextStyle: TextStyle(color: textPrimary),
      ),
    );
  }
}

File: kotlin/sample/app/src/main/java/com/edgeveda/example/Theme.kt
package com.edgeveda.example

import androidx.compose.ui.graphics.Color

/**
 * Centralized premium theme constants for the Veda app.
 * True black background with teal/cyan accent palette.
 * Matches the Flutter example app exactly.
 */
object AppTheme {
    // Core Colors
    val background = Color(0xFF000000)
    val surface = Color(0xFF0A0A0F)
    val surfaceVariant = Color(0xFF141420)
    val accent = Color(0xFF00BCD4)
    val accentDim = Color(0xFF00838F)

    // Text Colors
    val textPrimary = Color(0xFFF5F5F5)
    val textSecondary = Color(0xFF8A8A9A)
    val textTertiary = Color(0xFF5A5A6A)

    // UI Element Colors
    val border = Color(0xFF1E1E2E)
    val userBubble = Color(0xFF00838F)
    val assistantBubble = Color(0xFF0A0A0F)

    // Brand Colors
    val brandRed = Color(0xFFE50914)

    // Status Colors
    val danger = Color(0xFFEF5350)
    val success = Color(0xFF66BB6A)
    val warning = Color(0xFFFFB74D)
}